e2e/lib/k8up.bash

```diff
@@ -19,13 +30,82 @@ kustomize() {
 	go run sigs.k8s.io/kustomize/kustomize/v3 "${@}"
 }
 
+restic() {
+	kubectl run wrestic \
+		--rm \
+		--attach \
+		--restart Never \
+		--namespace "${DETIK_CLIENT_NAMESPACE-"k8up-system"}" \
+		--image "${WRESTIC_IMAGE-quay.io/vshn/wrestic}" \
+		--env "AWS_ACCESS_KEY_ID=myaccesskey" \
+		--env "AWS_SECRET_KEY=mysecretkey" \
+		--env "RESTIC_PASSWORD=myreposecret" \
+		--pod-running-timeout 10s \
+		--quiet=true \
+		--command -- \
+			restic \
+			--no-cache \
+			-r "s3:http://minio.minio.svc.cluster.local:9000/backup" \
+			"${@}" \
+			--json
+}
+
 prepare() {
 	mkdir -p "debug/${1}"
 	kustomize build "${1}" -o "debug/${1}/main.yml"
-	sed -i -e "s|\$E2E_IMAGE|'${E2E_IMAGE}'|" "debug/${1}/main.yml"
+	sed -i \
+		-e "s|\$E2E_IMAGE|'${E2E_IMAGE}'|" \
+		"debug/${1}/main.yml"
+	sed -i \
+		-e "s|\$WRESTIC_IMAGE|'${WRESTIC_IMAGE}'|" \
+		"debug/${1}/main.yml"
 }
 
 apply() {
 	prepare "${@}"
 	kubectl apply -f "debug/${1}/main.yml"
 }
+
+given_a_subject() {
+	kubectl create namespace k8up-e2e-subject || true
+	apply definitions/subject
+}
+
+is_minio_helm_repo_installed() {
+	echo -n "Is MinIO Repo available in Helm: "
+	helm repo list -o json |
+		jq -e 'map(select(.name == "minio")) | length == 1'
+	return ${?}
+}
+
+ensure_minio_helm_repo() {
+	if is_minio_helm_repo_installed; then
+		return
+	fi
+
+	helm repo add minio https://helm.min.io/
```

Comment:
Why not just call `helm repo add minio https://helm.min.io/` and `helm repo update` every time? I don't really see the need to do complex checking whether minio repo already exists etc and these commands are pretty fast.